tests/c/x64_imm_add.c

```diff
@@ -0,0 +1,60 @@
+// ignore-if: test $(uname -m) != x86_64
```

Comment:
I don't think a C test for this is useful (especially because C tests are *super* slow): a unit test in ykrt will cover all this better and *much* faster.